redaxo/src/addons/cronjob/lib/cronjob.php

```diff
@@ -15,7 +15,7 @@ abstract class rex_cronjob
     private $message = '';
 
     /**
-     * @param string $class
+     * @param class-string $class
      */
     final public static function factory($class)
     {
```

Comment:
Was ist das für ein case wo $class nicht autoloadable ist und direkt returned wird? Wann ist das notwendig?

Reply:
Ich glaube, wenn in der DB ein Cronjob steht zu einer Cronjob-Klasse aus einem Addon, und das Addon aber nicht mehr aktiviert ist.
Vermutlich wäre eigentlich `class_exists` besser.

Reply:
Aber warum wird dann $class returned?

Reply:
das fängt das addon an anderer stelle dann ab, denke ich. Ich glaube, es schreibt dann eine fehlermeldung ins cronjob-log, mit dem klassennamen. deswegen wird er hier als string zurückgegeben.
ich vermute mal, man könnte es besser mit einer exception hier umsetzen. dafür müsste man sich aber den code genauer anschauen, und die abfang-stelle anpassen.